.github/workflows/publish_snapshot_chart.yml

```diff
@@ -0,0 +1,47 @@
+name: Publish Snapshot Chart
+
+on:
+  schedule:
+  - cron: '0 0 * * *'
+
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'list of branches to publish (JSON)'
+        required: true
+        # keep in sync with default value of strategy matrix 'branch'
+        default: '["main"]'
+
+jobs:
+  publish-snapshot-chart:
+    if: ${{ startsWith(github.repository, 'apache/') }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # keep in sync with default value of workflow_dispatch input 'branch'
+        branch: ${{ fromJSON( inputs.branch || '["main"]' ) }}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ matrix.branch }}
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: 17
+        distribution: 'adopt'
+        cache: 'gradle'
+    - name: Build Operator
+      run: |
+        ./gradlew build spark-operator-api:relocateGeneratedCRD -x check --no-daemon
+    - name: Build Chart
+      run: |
+        cd build-tools/helm
+        mkdir charts
+        tar cvfz charts/spark-kubernetes-operator-0.1.0-SNAPSHOT.tgz spark-kubernetes-operator
```

Comment:
Is it possible to get the correct version `0.1.0-SNAPSHOT` automatically? If not, we may need to update this yml file after each release.

Reply:
Yes, we did `docker run spark-kubernetes-operator:$(./gradlew properties | grep '^version:' | awk '{print $2}')` in `build_and_test.yml`. For this one, let me improve it after we get a successful run, @viirya .